editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -1588,31 +1544,16 @@ fn not_artboard(document: &DocumentMessageHandler) -> impl Fn(&LayerNodeIdentifi
 }
 
 fn drag_shallowest_manipulation(responses: &mut VecDeque<Message>, selected: Vec<LayerNodeIdentifier>, tool_data: &mut SelectToolData, document: &DocumentMessageHandler) {
-	for layer in selected {
-		let ancestor = layer
-			.ancestors(document.metadata())
-			.filter(not_artboard(document))
-			.find(|&ancestor| document.network_interface.selected_nodes().selected_layers_contains(ancestor, document.metadata()));
-
-		let new_selected = ancestor.unwrap_or_else(|| layer.ancestors(document.metadata()).filter(not_artboard(document)).last().unwrap_or(layer));
-		tool_data.layers_dragging.retain(|layer| !layer.ancestors(document.metadata()).any(|ancestor| ancestor == new_selected));
-		tool_data.layers_dragging.push(new_selected);
+	// Only apply shallowest manipulation if we're in Shallow Select mode
+	if tool_data.nested_selection_behavior == NestedSelectionBehavior::Shallowest {
+		let selected: Vec<_> = selected.into_iter().filter(not_artboard(document)).collect();
+		if !selected.is_empty() {
+			responses.add(DocumentMessage::SetSelectedLayers { selected });
+		}
+	} else {
+		// In Deep Select mode, use the deepest manipulation logic
+		drag_deepest_manipulation(responses, selected, tool_data, document);
```

Comment:
It is rather confusing to have the `drag_shallowest_manipulation` call the `drag_deepest_manipulation`.